File: src/main/java/com/dennisturco/model/Quotation.java
package com.dennisturco.model;

import java.time.LocalDate;

import org.hibernate.annotations.Check;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Check(constraints = "vat_percentage >= 0 AND vat_percentage <= 100")
@Check(constraints = "discount_percentage >= 0 AND discount_percentage <= 100")
public class Quotation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @ManyToOne(optional = false)
    @JoinColumn(name = "quotation_status_id", nullable = false)
    private QuotationStatus quotationStatus;

    @Column(nullable = false, unique = true)
    private String number;

    @Column(nullable = false)
    private Float amount;

    @Column(nullable = false)
    private Integer vatPercentage;

    @Column(nullable = false)
    private Integer discountPercentage;

    @Column(length = 2000)
    private String description;

    @Column(length = 1000)
    private String notes;

    @Column(nullable = false)
    private LocalDate creationDate;

    @Column(nullable = false)
    private LocalDate lastUpdateDate;

    private LocalDate issueDate;
    private LocalDate validityDate;
}

File: GestioPro.Common/Models/Quotation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using GestioPro.Common.Enums;

namespace GestioPro.Common.Models;

[Table("quotations")]
public class Quotation
{
    [Key]
    public long Id { get; set; }

    [Required]
    public long CustomerId { get; set; }

    [ForeignKey(nameof(CustomerId))]
    public Customer Customer { get; set; } = null!;

    public QuotationStatus QuotationStatus { get; set; }

    [Required]
    public string Number { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    public required string Title { get; set; }

    [Required]
    [Range(0, float.MaxValue)]
    public float Amount { get; set; }

    [Required]
    [Range(0, 100)]
    public int VatPercentage { get; set; }

    [Required]
    [Range(0, 100)]
    public int DiscountPercentage { get; set; }

    [MaxLength(2000)]
    public string? Description { get; set; }

    [MaxLength(1000)]
    public string? Notes { get; set; }

    [Required]
    public DateTimeOffset CreationDate { get; set; }

    [Required]
    public DateTimeOffset LastUpdateDate { get; set; }

    public DateOnly? IssueDate { get; set; }
    public DateOnly? ValidityDate { get; set; }

    [Required]
    public bool IsDisabled { get; set; }
}
